src/gpu/sycl/ref_layer_normalizations.cpp

```diff
@@ -0,0 +1,245 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#include "gpu/sycl/ref_layer_normalizations.hpp"
+#include "common/c_types_map.hpp"
+#include "common/dnnl_traits.hpp"
+#include "gpu/sycl/layer_normalizations_kernels.hpp"
+#include "gpu/sycl/sycl_types.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace sycl {
+
+using namespace impl::sycl;
+
+status_t ref_layer_normalization_fwd_t::pd_t::init_conf() {
+    conf_ = sycl_layer_normalization_conf_t();
+
+    conf_.var_md
```

Comment:
I hit assert here when I run ` --lnorm --engine=gpu --dir=FWD_I --dt=f16:f16 --flags=CH --inplace=true 30x300`.
Both mds are zero mds for such case. You should also check that `desc_.prop_kind == prop_kind::forward_training`.